test/ctia/entity/incident_test.clj

```diff
@@ -106,18 +106,20 @@
    "High" 3
    "Critical" 4})
 
-(defn gen-new-incident [severity]
-  (let [order (ctim-severity-order severity)
-        _ (if (some? severity)
-            (assert (number? order)
-                    (str "Unmapped severity " (pr-str severity)))
-            (assert ((some-fn nil? number?) order)))]
-    (-> new-incident-minimal
-        (dissoc :id :severity)
-        ;; test missing severity if nil
-        (cond-> (some? order) (assoc :severity severity))
-        (assoc :title (str (java.util.UUID/randomUUID))
-               :revision (or order 0)))))
+(defn gen-new-incident
+  ([] (gen-new-incident "High"))
+  ([severity]
+   (let [order (ctim-severity-order severity)
+         _ (if (some? severity)
+             (assert (number? order)
+                     (str "Unmapped severity " (pr-str severity)))
+             (assert ((some-fn nil? number?) order)))]
```

Comment:
`ctim-severity-order` is a map of string-> number, I can't imagine a situation when it could return something different from nil or number. Then instead of assert it could be a bit more explicit `(or (ctim-severity-order severity) (throw (ex-info (str "Unmapped severity" severity) {})))`

Reply:
100% agree, that's why it's an assertion :)

Reply:
but then, why is assertion needed? you wouldn't assert that `str` always returns a string. How is it different from the mapping from string to number? I don't mean to be picky, but I still don't understand why these four lines of code with branching logic are needed.